src/main/java/io/cdap/plugin/gcp/gcs/GCSPath.java

```diff
@@ -93,13 +93,14 @@ public static GCSPath from(String path) {
 
   private static URI getURI(String path) {
     try {
-      URI uri = new URI(path);
+      String modifiedPath = path.replaceAll("\\s", "%20");
```

Comment:
Don't do this manual replacement. Use `URLEncoder.encode(path, "UTF-8")` instead.
Also, `\\s` includes all white spaces (e.g. tab), which should be encoded as `%09`.

Reply:
Correction, please use `UrlEscapers.urlPathSegmentEscaper().escape(path)` instead.